utils/src/main/scala/za/co/absa/enceladus/utils/broadcast/BroadcastUtils.scala

```diff
@@ -124,305 +166,349 @@ object BroadcastUtils {
     errorUdf
   }
 
-  private def getMappingLambdaParam1(mappingTable: Broadcast[LocalMappingTable], defaultValueOpt: Option[Any]): AnyRef = {
-    defaultValueOpt match {
-      case None =>
-        param1: Any => {
-          val mt = mappingTable.value.map
-          mt.getOrElse(Seq(param1), null)
+  private def mapMultipleOutputs(mappingTable: Broadcast[LocalMappingTable], row: Option[Any], defaults: Map[String, Any]): Row = {
+    row match {
+      case Some(row: Row) => {
+        val rowValues: Map[String, Any] = row.getValuesMap[Any](mappingTable.value.outputColumns.keys.toSeq)
+        val values = rowValues.map{ case (field, value) => {
+          if (value == null) defaults(field)
+          else value
+        }}
+        new GenericRowWithSchema(values.toArray, row.schema)
+      }
+      case Some(_) => null
+      case None => {
+        if (defaults.isEmpty) null
+        else {
+          val values = mappingTable.value.outputColumns.values.toSeq.map(field => defaults.getOrElse(field, null))
+          Row.fromSeq(values)
         }
-      case Some(defaultValue) =>
+      }
+    }
+  }
+
+  private def getMappingLambdaParam1(mappingTable: Broadcast[LocalMappingTable],
```

Comment:
@AdrianOlosutean hey let's have a call/pair programming session over:
* `private def getMappingLambdaParamX(_, _, false)`
* `private def mapMultipleOutputs`
* `final case class LocalMappingTable`
I would like to discus some changes, imho improvements, but maybe not. And too long to describe and discuss via chat here.